pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -81,16 +82,32 @@
         );
     }
 
+    @Override
+    public CompletableFuture<List<CompletableFuture<SchemaAndMetadata>>> getAllSchemas(String schemaId) {
+        return schemaStorage.getAll(schemaId).thenApply(schemas ->
+                schemas.stream().map(future -> future.thenCompose(stored ->
+                    Functions.bytesToSchemaInfo(stored.data)
+                        .thenApply(Functions::schemaInfoToSchema)
+                        .thenApply(schema -> new SchemaAndMetadata(schemaId, schema, stored.version))
+        )).collect(Collectors.toList()));
+    }
+
     @Override
     @NotNull
     public CompletableFuture<SchemaVersion> putSchemaIfAbsent(String schemaId, SchemaData schema,
                                                               SchemaCompatibilityStrategy strategy) {
         return getSchema(schemaId)
             .thenApply(
                 (existingSchema) ->
-                    existingSchema == null
-                        || existingSchema.schema.isDeleted()
-                        || isCompatible(existingSchema, schema, strategy))
+                {
+                    try {
+                        return existingSchema == null
+                            || existingSchema.schema.isDeleted()
+                            || (isCompatible(schemaId, schema, strategy).get());
```

Comment:
I don't think it is a good idea to use blocking `get` here.

The whole logic can be rewritten as

```
return getSchema(schemaId)
    .thenCompose(
        (existingSchema) -> {
            if (existingSchema == null || existingSchema.schema.isDeleted()) {
                return completedFuture(true);
            } else {
                return isCompatible(schemaId, schema, strategy);
            }
        }
    }
    .thenCompose(...)
```